moabb/evaluations/splitters.py

```diff
@@ -101,3 +109,112 @@ def split(self, y, metadata):
                 for train_ix, test_ix in splitter.split(indices, y_session):
 
                     yield indices[train_ix], indices[test_ix]
+
+
+class CrossSessionSplitter(BaseCrossValidator):
+    """Data splitter for cross session evaluation.
+
+    Cross-session evaluation uses a Leave-One-Session-Out cross-validation to
+    evaluate performance across sessions, but for a single subject. This splitter
+    assumes that all data from all subjects is already known and loaded.
+
+    .. image:: images/crosssess.jpg
+        :alt: The schematic diagram of the CrossSession split
+        :align: center
+
+    The inner cross-validation strategy can be changed by passing the
+    `cv_class` and `cv_kwargs` arguments. By default, it uses LeaveOneGroupOut,
+    which effectively performs Leave-One-Session-Out cross-validation when
+    sessions are passed as groups.
+
+    Parameters
+    ----------
+    shuffle : bool, default=False
+        Whether to shuffle the session order for each subject.
+    random_state: int, RandomState instance or None, default=None
+        Controls the randomness when `shuffle` is True.
+        Pass an int for reproducible output across multiple function calls.
+    cv_class: cross-validation class, default=LeaveOneGroupOut
+        Inner cross-validation strategy for splitting the sessions.
+        For cross-session splitting, LeaveOneGroupOut is the most suitable as default.
+    cv_kwargs: dict
+        Additional arguments to pass to the inner cross-validation strategy.
+    """
+
+    def __init__(
+        self,
+        shuffle: bool = False,
+        random_state: int = None,
+        cv_class: type[BaseCrossValidator] = LeaveOneGroupOut,
+        **cv_kwargs: dict,
+    ):
+        self.cv_class = cv_class
+        self.cv_kwargs = cv_kwargs
+        self._cv_kwargs = dict(**cv_kwargs)
+
+        self.shuffle = shuffle
+        self.random_state = random_state
+
+        self._rng = check_random_state(random_state) if shuffle else None
```

Comment:
If, after the review, `self._rng` is not needed anymore, I would just pass the shuffle argument to the cv_class

Reply:
The `rng` is needed to seed the various `cv_class`.

Reply:
True! I forgot we were initializing one cv per subject...

However, I just realise that initializing `rng` in the `__init__` is not correct, because if we call multiple times the same `CrossSessionSplitter` instance, it will start with a different random state each time. Instead, `rng` should be initialized at the beginning of the `split` method

Reply:
Fair point :)